include/libtorrent/io.hpp

```diff
@@ -80,11 +81,16 @@ namespace libtorrent
 				|| std::is_enum<In>::value>::type>
 		inline void write_impl(In data, OutIt& start)
 		{
+			using OutItT = typename std::iterator_traits<OutIt>::value_type;
+			using Byte = typename std::conditional<
+				std::is_same<OutItT, void>::value, char, OutItT>::type;
```

Comment:
I take it this is to avoid signed <-> unsigned conversions, is it? It feels a bit questionable to use bit-wise operations on signed integers. doesn't ``static_cast<>`` suppress the conversion warning?